default-go-test.mk

```diff
@@ -24,11 +25,13 @@ unit: test
 test: ## Run unit tests (go test)
 ifeq ($(CONTAINER_RUNNABLE), 0)
 		$(CONTAINER_RUNTIME) run -it -v "$(CURDIR):/go/src" -w /go/src docker.io/library/golang:${GO_VERSION}-alpine3.17 \
-         sh -e -c "go test ./... -v -coverprofile ${TEST_COVERAGE_FILE}; \
+         sh -e -c "go test ./... -v -coverprofile ${TMP_TEST_COVERAGE_FILE}; \
```

Comment:
do we need to add this file?

Reply:
I did this to strip the fake_client.go from the coverage report. I guess I could name it with a _test.go suffix.
The report also doesn't really give a proper reflection of the overall coverage. Do we need an _test.go in each reconciler dir so that a report is produced?

Reply:
if we have to change the report we should do this in another PR. in go you normally put a _test.go file next to the functions. Now if we can improve the reporting we can do but do in another PR.

Reply:
Done